src/Native/Unix/configure.cmake

```diff
@@ -174,6 +174,20 @@ check_cxx_source_compiles(
 
 check_cxx_source_compiles(
     "
+    #include <dirent.h>
+    int main(void)
+    {
+        DIR* dir;
+        struct dirent* entry;
+        struct dirent* result;
+        readdir_r(dir, entry, &result);
```

Comment:
With the current build options, stop building if any deprecated functions are used, it probably does what I expect. The test environment I have been using is the dotnet-clr AUR package.
I could double triple check for you if you would like.